kivy app/main.py

```diff
@@ -0,0 +1,163 @@
+# Filename: main.py
+import random
+import time
+from kivy.app import App
+from kivy.uix.widget import Widget
+from kivy.properties import NumericProperty
+from kivy.clock import Clock
+from kivy.uix.popup import Popup
+from kivy.uix.label import Label
+from kivy.uix.button import Button
+
+# TO DO LIST:
+# If not using pages then menu screens/high score widgets
+# Add images/sprites
+# Change truck speeds (different levels)
+# Leader board client server code
+
+RUNTIME = 30  # maybe move & change depending on level
+LIVES = 3
+SCORE = 0
+
+
+class PlayerObject(Widget):
+    """ Player controlled object """
+    score = NumericProperty(SCORE)
+    lives = NumericProperty(LIVES)
+
+    def truck_collision(self, truck):
+        """ The function minuses 1 from the current lives value when an instance of Player collides
+        with an instance of Trucks
+        :param truck:
+        :return:
+        """
+        if self.collide_widget(truck):
+            truck.x = -750
+            if self.lives > 0:
+                self.lives -= 1
+            elif self.lives <= 0:
+                print "Game Over"
+                # TheGame.end_game()
+                # self.remove_widget(Trucks)
+
+
+class Trucks(Widget):
+    def move(self):
+        """ This function adds the value of TRUCK_SPEED to an instance of Trucks x coordinate everytime it's called
+        :return:
+        """
+        if self.center_x > 480:
+            self.center_x = 0
+        else:
+            self.center_x += self.speed
+
+
+class TheGame(Widget):
+    player = PlayerObject()
+    timer = NumericProperty(RUNTIME)
+    level = NumericProperty(1)
+    speed = NumericProperty(0)
+
+    traffic_list = []
+
+    end = Label()
+    help = Button()
+
+    def traffic(self, traffic_list):
+        """This function generates instances of the Trucks objects and adds it to traffic_list. Each instance
+        has a random X value and a Y value randomly chosen from a list
+        :param traffic_list:
+        :return:
+        """
+        for t in range(0, 10):
+            truck = Trucks()
+            y_options = [100, 200, 300, 400, 500]
+            y_choice = random.choice(y_options)
+            truck.center_x = random.randint(1, 500)
+            truck.center_y = y_choice
+            self.add_widget(truck)
+            self.traffic_list.append(truck)
+        return traffic_list
+
+    def the_timer(self, timer):
+        """ This function minus one from the current timer value whenever called.
+        :param timer:
+        :return:
+        """
+        self.timer -= 1
+        if self.level == 1:
+            self.speed = 2
```

Comment:
Just noticed that this function is called every second. In this case, it's probably best to make changes to the speed at the time that the level changes, rather than in here.